### PR TITLE
feat: ✨  add `include_diabetes_diagnoses()` with tests

### LINKED ISSUE #91: Create function `include_diabetes_diagnoses()`
From @Aastedet 's original issue #6 :
> At a point we may have to address differences between LPR2 and LPR3 data, unless we delegate that to the users' preprocessing of raw data.

### PR DESCRIPTION
**Pull request by Aastedet**
This PR adds:

- The logic for filtering diabetes diagnoses in lpr2 and lpr3 to algorithm.csv
- The include_diabetes_diagnoses() function + tests.
   - I changed the way department specialty is defined in LPR3 from exactly matching a string vector to be regex-based instead. It's not 100% identical to the previous implementation, but it doesn't have to be, as it wasn't a gold standard, and most of the departments specialties are edge cases anyway (type-specific diabetes primary diagnoses mainly come from a few specialties). This way is more succinct and should be more robust to register data shenanigans.
   - The data in the tibbles for testing I've added might not cover 100% of all edge cases, but it's a good start at least. I'm not sure if the included tests are the best implementation, but they somewhat cover my eyeballing of the output.
- A helper function, get_majority_t1d_diagnoses(), containing the logic defining majority of diabetes diagnoses + tests for this function

Closes #91 

## Checklist

- [ ] Haven't `just run-all`

### CODE REVIEW

File: R/include-diabetes-diagnoses.R [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+  # Filter to diabetes diagnoses using expression filter.
+  lpr3_diabetes_diagnoses <- diagnoser |>
+    column_names_to_lower() |>
+    dplyr::filter(!!lpr3_criteria)
+
+  lpr3_diabetes <- join_lpr3(kontakter = kontakter, diagnoser = lpr3_diabetes_diagnoses)
+
+  # Define variables
+  hovedspeciale_other_medical_pattern <- '[Mm]edicin|[Gg]eriatri|[Hh]epatologi|[Hh]æmatologi|[Ii]nfektion|[Kk]ardiologi|[Nn]efrologi|[Rr]eumatologi|[Dd]ermato|[Nn]eurologi|[Oo]nkologi|[Oo]ftalmologi|[Nn]eurofysiologi'
```
Comment (Aastedet) [inline R/include-diabetes-diagnoses.R:70]: This is the regex I mentioned

File: R/include-diabetes-diagnoses.R [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+#' \dontrun{
+#' include_diabetes_diagnoses(lpr_diag, lpr_adm, diagnoser, kontakter)
+#' }
+include_diabetes_diagnoses <- function(lpr_diag, lpr_adm, diagnoser, kontakter) {
+  verify_required_variables(register_data$lpr_diag, "lpr_diag")
+  verify_required_variables(register_data$lpr_adm, "lpr_adm")
+  verify_required_variables(register_data$diagnoser, "diagnoser")
+  verify_required_variables(register_data$kontakter, "kontakter")
+
+  # Filter and join LPR2 data:
```
Comment (signekb) [inline R/include-diabetes-diagnoses.R:31]:
```suggestion
  # Filter and join LPR2 data
```

File: R/include-diabetes-diagnoses.R [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  lpr2_criteria <- get_algorithm_logic("lpr2_diabetes") |>
+    # To convert the string into an R expression.
+    rlang::parse_expr()
+
+  # Filter to diabetes diagnoses using expression filter.
```
Comment (signekb) [inline R/include-diabetes-diagnoses.R:36]:
```suggestion
  # Filter to diabetes diagnoses using expression filter
```

File: R/include-diabetes-diagnoses.R [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+  lpr2_diabetes_diagnoses <- lpr_diag |>
+    column_names_to_lower() |>
+    dplyr::filter(!!lpr2_criteria)
+
+  lpr2_diabetes <- join_lpr2(lpr_adm = lpr_adm, lpr_diag = lpr2_diabetes_diagnoses)
```
Comment (signekb) [inline R/include-diabetes-diagnoses.R:41]: Maybe it's fine just calling this variable `lpr2`?

File: R/include-diabetes-diagnoses.R (hunk 1 above)
Comment (signekb) [reply R/include-diabetes-diagnoses.R:70]: I'm thinking that it makes sense to move this to `algorithm.csv` and maybe adding the potentially capitalised letters in #190 👍

File: R/include-diabetes-diagnoses.R [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+      n_t1d_medical = sum(
+        (is_t1d == TRUE) &
+          (is_primary == TRUE) & (department == 'other medical'),
+        na.rm = TRUE
+      ),
+      n_t2d_medical = sum(
+        (is_t2d == TRUE) &
+          (is_primary == TRUE) & (department == 'other medical'),
+        na.rm = TRUE
+      )
```
Comment (signekb) [inline R/include-diabetes-diagnoses.R:110]: Could/should this be moved to some simple helper/filter functions?

**Review by signekb — CHANGES_REQUESTED** [review]
Nice work 🎉  
Just a mini review with some initial comments/thoughts :)

File: R/include-diabetes-diagnoses.R (hunk 5 above)
Comment (Aastedet) [reply R/include-diabetes-diagnoses.R:110]: Hmm, yes... something like get_n_type_specific_diagnoses()

It could also be made more succinct by removing all the " == TRUE" parts, since they're not needed. I originally didn't write them out, but I ended up adding the for readability (but now I think they degrade it).

File: R/include-diabetes-diagnoses.R (hunk 1 above)
Comment (Aastedet) [reply R/include-diabetes-diagnoses.R:70]: Once #221 gets merged to main, I'll add this to `algorithm.R` and update the code here accordingly.

File: R/include-diabetes-diagnoses.R (hunk 5 above)
Comment (Aastedet) [reply R/include-diabetes-diagnoses.R]: I did this with the latest commit.

**Comment by Aastedet** [comment]
This PR will be rewritten to match the new way of generating synthetic data and extracting algorithm logic.

**Comment by Aastedet** [comment]
Yeah. That probably makes more sense. I'll delete this and create a new one when I finish the final things.
